WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -604,34 +606,22 @@ public boolean onOptionsItemSelected(MenuItem item) {
             return false;
         }
 
-        // Disable ActionBar buttons while there are failed media uploads in the post/page
+        // Show an Alert Dialog asking the user if he wants to remove all failed media before upload
         if (mEditorFragment.hasFailedMediaUploads()) {
-            ToastUtils.showToast(this, R.string.editor_toast_failed_uploads, Duration.SHORT);
-            return false;
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setMessage(R.string.editor_toast_failed_uploads)
+                    .setPositiveButton(R.string.editor_remove_failed_uploads, new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int id) {
+                            // Clear failed uploads
+                            mEditorFragment.removeAllFailedMediaUploads();
```

Comment:
We could retry the save action here, but the user won't be able to verify his post looks good after the removal of failed media.